### PR TITLE
[mypyc] Fix ParamSpec

### PR DESCRIPTION
**Pull request by hauntsaninja**
Fixes https://github.com/mypyc/mypyc/issues/1051

### CODE REVIEW

File: mypyc/irbuild/mapper.py [hunk 1]
```diff
@@ -123,6 +124,8 @@ def type_to_rtype(self, typ: Type | None) -> RType:
                 return tuple_rprimitive
         elif isinstance(typ, CallableType):
             return object_rprimitive
+        elif isinstance(typ, ParamSpecType):
+            return object_rprimitive
```
Comment (AlexWaygood) [inline mypyc/irbuild/mapper.py:128]: If it's `P.args` or `P.kwargs`, we use `tuple[object, ...]` and `dict[str, object]` as the upper bound of the ParamSpecType (respectively). Does that imply we should do something similar to the TypeVarType branch below?

```suggestion
            return self.type_to_rtype(typ.upper_bound)
```

**Comment by AlexWaygood** [comment]
Looks like TypeVarTuple might also not be handled — worth adding that too?

File: mypyc/irbuild/mapper.py (hunk 1 above)
Comment (hauntsaninja) [reply mypyc/irbuild/mapper.py]: Good point, thanks!

**Review by AlexWaygood — APPROVED** [review]

**Comment by kai3341** [comment]
I've install mypy directly from this branch to test on my code and found it throws different error now:
```
Traceback (most recent call last):
  File "/home/kai/Projects/ovpn-mgmt/build/setup.py", line 5, in <module>
    ext_modules=mypycify(['./ovpn-mgmt.py'], opt_level="3", debug_level="1"),
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/build.py", line 525, in mypycify
    groups, group_cfilenames = mypyc_build(
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/build.py", line 435, in mypyc_build
    group_cfiles, ops_text = generate_c(
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/build.py", line 243, in generate_c
    modules, ctext = emitmodule.compile_modules_to_c(
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/codegen/emitmodule.py", line 428, in compile_modules_to_c
    modules = compile_modules_to_ir(result, mapper, compiler_options, errors)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/codegen/emitmodule.py", line 272, in compile_modules_to_ir
    scc_ir = compile_scc_to_ir(trees, result, mapper, compiler_options, errors)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/codegen/emitmodule.py", line 227, in compile_scc_to_ir
    modules = build_ir(scc, result.graph, result.types, mapper, compiler_options, errors)
  File "/usr/lib/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/irbuild/main.py", line 95, in build_ir
    transform_mypy_file(builder, module)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/irbuild/main.py", line 140, in transform_mypy_file
    builder.accept(node)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/irbuild/builder.py", line 266, in accept
    node.accept(self.visitor)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypy/nodes.py", line 1181, in accept
    return visitor.visit_class_def(self)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/irbuild/visitor.py", line 168, in visit_class_def
    transform_class_def(self.builder, cdef)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/irbuild/classdef.py", line 138, in transform_class_def
    cls_builder.add_method(get_func_def(stmt))
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/irbuild/classdef.py", line 270, in add_method
    handle_ext_method(self.builder, self.cdef, fdef)
  File "/home/kai/Projects/ovpn-mgmt/.venv/lib/python3.11/site-packages/mypyc/irbuild/function.py", line 458, in handle_ext_method
    class_ir.method_decls[name].sig, base.method_decls[name].sig
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^
ovpn-mgmt.py:1161: KeyError: 'handle1234567'
```

I've just renamed my method `handle` to `handle1234567` to be sure why does mypyc throws an exception.

So, what should I do to help you? Is is separated issue or this issue continuation? I'll try to minimize the code to reproduce this issue

**Comment by hauntsaninja** [comment]
Sounds like a separate issue, searching stacktrace sounds similar to https://github.com/mypyc/mypyc/issues/767. The best way to help is to open a PR, I don't have time to look into it myself (and don't really know much about mypyc)

**Review by JukkaL — APPROVED** [review]

**Pull request merged**